bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/robot/RRMapDraw.java

```diff
@@ -160,7 +165,8 @@ private void drawMap(Graphics2D g2d, float scale) {
                                 g2d.setColor(Color.BLACK);
                                 break;
                             case 7:
-                                g2d.setColor(ROOM_COLORS[Math.round(mapId / 2)]);
+                                g2d.setColor(ROOM_COLORS[mapId % 15]);
```

Comment:
Is there a reason you are only allowing 15 colors instead of the full 16?
```suggestion
                                g2d.setColor(ROOM_COLORS[mapId % ROOM_COLORS.length]);
```

Reply:
yes, I saw that in quite a few occasions somehow both room 1 & 17 were defined. (without many in between)
That made the color the same for both rooms, hence not recognizable as separate rooms anymore.
so this is little hack to avoid that... double colors can obviously still occur, but this seems to reduce the chance.

Reply:
Can't you just make sure that each room has a unique color?

Reply:
not easy to find 32 distinct colors while keeping the path, walls, vacuum etc still properly visible.
The Xiaomi app also does not do that.. don't know exactly how many they are using, but certainly not 32.